pkg/descheduler/strategies/topologyspreadconstraint.go

```diff
@@ -0,0 +1,242 @@
+					if err != nil {
+						klog.ErrorS(err, "couldn't parse label selector as selector", "selector", constraint.LabelSelector)
+					}
+					if !s.Matches(labels.Set(pod.Labels)) {
+						continue
+					}
+
+					// 5. If the pod's node matches this constraint's topologyKey, create a topoPair and add the pod
+					node, err := client.CoreV1().Nodes().Get(ctx, pod.Spec.NodeName, metav1.GetOptions{})
+					if err != nil {
+						klog.ErrorS(err, "couldn't get node", "node", pod.Spec.NodeName)
+					}
+					nodeValue, ok := node.Labels[constraint.TopologyKey]
+					if !ok {
+						continue
+					}
+					// 6. create a topoPair with key as this TopologySpreadConstraint
+					topoPair := topologyPair{key: constraint.TopologyKey, value: nodeValue}
+					// 7. add the pod with key as this topoPair
+					constraintTopologies[topoPair] = append(constraintTopologies[topoPair], &pod)
+					sumPods++
+				}
+
+				// To actually determine how many pods need to be moved, we sort the topologies in ascending length
+				// Then, split the topologies in half around the ideal average (ie, perfect spreading)
+				// Sort the list of topologies with length <= the average in ascending order, and all > the average descending.
+				// For example, the list of topologies with these sizes (avg=5):
+				//  2
+				//  5
+				//  3
+				//  8
+				//  5
+				//  7
+				// Would end up in 2 lists like:
+				//  2  8
+				//  3  7
+				//  5
+				//  5
+				// We then start at [0,0] and compare the 2 topology sizes.
+				// If the diff of the size of the topologies is more than the maxSkew, we will move up to half that skew,
+				// or the available pods from the "aboveAvg" topology, whichever is lower.
+				// (Note, we will only move as many pods from a topology as possible without bringing it below the ideal average.)
+				// If the diff is within the skew, we move to the next "belowAvg" topology.
+				// If the "aboveAvg" topology can't give any more without falling below the average, we move to the next "aboveAvg"
+				//
+				// Following this, the above topologies end up "sorted" as:
+				//  5  5
+				//  5  5
+				//  5
+				//  5
+				// (assuming even distribution by the scheduler of the evicted pods)
+				idealAvg := int(math.Ceil(sumPods / float64(len(constraintTopologies))))
+				belowAvg, aboveAvg := splitTopologiesAroundAverage(constraintTopologies, idealAvg)
+
+				// i is the index for belowAvg
+				// j is the index for aboveAvg
+				i, j := 0, 0
+				for i < len(belowAvg) && j < len(aboveAvg) {
+					// if j has no more to give without falling below the ideal average, move to next aboveAvg
+					if len(aboveAvg[j].pods)-idealAvg <= 0 {
+						j++
+					}
+
+					// skew = diff - maxSkew
+					skew := math.Abs(float64(len(belowAvg[i].pods)-len(aboveAvg[j].pods))) - float64(constraint.MaxSkew)
+
+					// if i and j are within the maxSkew of each other, move to next belowAvg
+					if int32(skew) <= constraint.MaxSkew {
```

Comment:
`skew` already has `constraint.MaxSkew` substracted from itself. Can you more elaborate on what's the test checking to reflect the condition?

Reply:
Good catch, I'm doubly-accounting for the `maxSkew` from what I described above (in https://github.com/kubernetes-sigs/descheduler/pull/413#discussion_r499789794).

In this case, if I wanted to do what I said there, this check should be `if int32(skew) <= 0`. But removing that subtraction and keeping this check makes more sense imo